File: Лабораторна №5/laba5py/Work_with_vectors.py
from TVector import  R2, R3
import random

def create_arr_of_vectors(n, m):
    arr = []
    mode = input("Random or manually? Random - enter 'r', manually - enter 'm': ")
    while mode != 'r' and mode !='m':
        mode = input("Try again! ")
    if mode == 'r':
        limits = input("Enter range (first min than max value): ").split()
        rangeMin = int(limits[0])
        rangeMax = int(limits[1])       
        for i in range(n):
            x = random.randint(rangeMin, rangeMax)
            y = random.randint(rangeMin, rangeMax)
            arr.append(R2(x, y))
        for i in range(m):
            x = random.randint(rangeMin, rangeMax)
            y = random.randint(rangeMin, rangeMax)
            z = random.randint(rangeMin, rangeMax)
            arr.append(R3(x, y, z))
    else:
        print("R2:")
        for i in range(n):
            values = input("x, y: ").split()
            x = int(values[0])
            y = int(values[1])
            arr.append(R2(x, y))
        for i in range(m):
            values = input("x, y, z: ").split()
            x = int(values[0])
            y = int(values[1])
            z = int(values[2])
            arr.append(R3(x, y, z))
    return arr

#make one function

def display_vector(arr):
    for i in arr:
        i.display()
        print(f"| length = {i.calc_length()}")

def calc_sum_of_parallels(arr, n):
    arr_par = []
    sum_length = 0
    for i in range(1, n):
        if arr[0].is_parallel(arr[i]):
            arr_par.append(arr[i])
            sum_length += arr[i].calc_length()
    return sum_length, arr_par

def calc_sum_of_perpendicular(arr, n, m):
    arr_perp = []
    sum_length = 0
    for i in range(n + 1, n + m):
        if arr[n].is_perpendicular(arr[i]):
            arr_perp.append(arr[i])
            sum_length += arr[i].calc_length()
    return sum_length, arr_perp

File: Лабораторна №5/laba5cpp/Work_with_vectors.cpp
#include <iostream>
#include <vector>
#include "TVector.h"
#include "Work_with_vectors.h"

using namespace std;

void create_arr_of_vectors(vector <TVector*>& arr, int n, int m)
{
	cout << "Random or manually? Random - enter 'r', manually - enter 'm'\n";
	char mode;
	cin >> mode;
	while (mode != 'r' && mode != 'm')
	{
		cout << "Try again! ";
		cin >> mode;
	}
	if (mode == 'r')
	{
		int rangeMin, rangeMax;
		cout << "Enter range (first min than max value): ";
		cin >> rangeMin;
		cin >> rangeMax;
		for (int i = 0; i < n; i++)
			arr.push_back(new R2(rangeMin, rangeMax));
		for (int i = 0; i < m; i++)
			arr.push_back(new R3(rangeMin, rangeMax));
	}
	else
	{
		double x, y, z;
		cout << "R2:\n";
		for (int i = 0; i < n; i++)
		{
			cout << "x, y: ";
			cin >> x; 
			cin >> y;
			arr.push_back(new R2(x, y));
		}
		cout << "R3:\n";
		for (int i = 0; i < m; i++)
		{
			cout << "x, y, z: ";
			cin >> x;
			cin >> y;
			cin >> z;
			arr.push_back(new R3(x, y, z));
		}
	}
}
void display_arr_of_vectors(vector <TVector*> arr, int n, int m)
{
	cout << "R2 vectors: \n";
	for (int i = 0; i < n; i++)
	{
		arr[i]->display();
		cout << "| length = " << arr[i]->calc_length() << "\n";		
	}
	cout << "R3 vectors: \n";
	for (int i = n; i < n + m; i++)
	{
		arr[i]->display();
		cout << "| length = " << arr[i]->calc_length() << "\n";
	}
}
void display_vector(vector <TVector*> arr)
{
	for (int i = 0; i < arr.size(); i++)
	{
		arr[i]->display();
		cout << "| length = " << arr[i]->calc_length() << "\n";
	}
}

double calc_sum_of_parallels(vector <TVector*> arr, int n, int m, vector <TVector*>& arrPar)
{
	double sum = 0;
	for (int i = 1; i < n; i++)
	{
		if (arr[0]->is_parallel(arr[i]))
		{
			sum += arr[i]->calc_length();
			arrPar.push_back(arr[i]);
		}			
	}
	return sum;
}
double calc_sum_of_perpendicular(vector <TVector*> arr, int n, int m, vector <TVector*>& arrPerp)
{
	double sum = 0;
	for (int i = n + 1; i < n + m; i++)
	{
		if (arr[n]->is_perpendicular(arr[i]))
		{
			sum += arr[i]->calc_length();
			arrPerp.push_back(arr[i]);
		}
	}
	return sum;
}